src/lib/Decentraland.ts

```diff
@@ -226,16 +232,44 @@ export class Decentraland extends EventEmitter {
     return this.wallet?.getAddress()!
   }
 
-  async getAddressAndSignature(messageToSign: string): Promise<LinkerResponse> {
+  async getAddressAndSignature(
+    messageToSign: string
+  ): Promise<LinkerResponseScenenDeploy> {
     if (this.wallet) {
       const [signature, address] = await Promise.all([
         this.wallet.signMessage(messageToSign),
         this.wallet.getAddress()
       ])
-      return { signature, address }
+      return {
+        responseType: 'scene-deploy',
+        payload: { signature, address, chainId: ChainId.ETHEREUM_MAINNET }
+      }
+    }
+
+    const response = await this.link(messageToSign)
+    if (response.responseType !== 'scene-deploy') {
+      console.error({ response })
+      throw new Error('Linker-dapp responses an invalid scene sign')
     }
+    return response
+  }
 
-    return this.link(messageToSign)
+  async getIdentity(): Promise<LinkerResponseIdentity> {
```

Comment:
if this fn does the same as `getAddressAndSignature(message: string`, why no reuse it ?

```suggestion
  async getAddressAndSignature(message?: string, type: 'identity' | 'scene-deploy'): Promise<LinkerResponseIdentity> {
  
  }
```

Reply:
almost